### ISSUE TITLE
should set training mode explicitly

### ISSUE THREAD

**Issue by benwu232** [ncullen93/torchsample#44]
In module_trainer.py, it is better to set training mode as True in fit and fit_loader, and False in predict and predict_loader.

In current code, when I set a break point in predict_loader(), I found the self.model.training == True.  This can impact the prediction performance and even cause memory leak when I use my own noise module (which should only run in training mode).

So I suggest add self.model.train(mode=True) at the first line of those "fit" functions, and self.model.train(mode=False) at the first line of those "predict" functions.

**Comment by ncullen93** [comment]
ok i did this is new version. thank you :) . Still trying to figure out how to best do this.
